<a href="https://colab.research.google.com/github/pjcroxas/Digital-Marketing/blob/main/Analyzing_Advertisement_Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import datetime as dt
from datetime import datetime

In [ ]:
dir = "/content/drive/My Drive/Data Science Portfolio/2024/Digital Marketing/CTR in Advertisement/data"
ctr_df = pd.read_csv(dir+"/Ad_click_prediction_train.csv")
ctr_df.head()

session_id          DateTime  user_id product  campaign_id  webpage_id  \
0      140690  2017-07-02 00:00   858557       C       359520       13787   
1      333291  2017-07-02 00:00   243253       C       105960       11085   
2      129781  2017-07-02 00:00   243253       C       359520       13787   
3      464848  2017-07-02 00:00  1097446       I       359520       13787   
4       90569  2017-07-02 00:01   663656       C       405490       60305   

   product_category_1  product_category_2  user_group_id  gender  age_level  \
0                   4                 NaN           10.0  Female        4.0   
1                   5                 NaN            8.0  Female        2.0   
2                   4                 NaN            8.0  Female        2.0   
3                   3                 NaN            3.0    Male        3.0   
4                   3                 NaN            2.0    Male        2.0   

   user_depth  city_development_index  var_1  is_click  
0         3.0                     3.0      0         0  
1         2.0                     NaN      0         0  
2         2.0                     NaN      0         0  
3         3.0                     2.0      1         0  
4         3.0                     2.0      1         0

In [ ]:
ctr_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 463291 entries, 0 to 463290
Data columns (total 15 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   session_id              463291 non-null  int64  
 1   DateTime                463291 non-null  object 
 2   user_id                 463291 non-null  int64  
 3   product                 463291 non-null  object 
 4   campaign_id             463291 non-null  int64  
 5   webpage_id              463291 non-null  int64  
 6   product_category_1      463291 non-null  int64  
 7   product_category_2      97437 non-null   float64
 8   user_group_id           445048 non-null  float64
 9   gender                  445048 non-null  object 
 10  age_level               445048 non-null  float64
 11  user_depth              445048 non-null  float64
 12  city_development_index  338162 non-null  float64
 13  var_1                   463291 non-null  int64  
 14  is_click            

In [ ]:
print("No. of users: ", ctr_df["user_id"].nunique())
print("No. of products: ", ctr_df["product"].value_counts())
print("No. of product category: ", ctr_df["product_category_1"].nunique())
print("No. of advertisement: ", ctr_df["campaign_id"].nunique())
print("No. of wepbage: ", ctr_df["webpage_id"].nunique())

No. of users:  150347
No. of products:  product
C    163501
H    109574
I     63711
D     41064
B     22479
E     21452
A     15391
J      9698
G      9414
F      7007
Name: count, dtype: int64
No. of product category:  5
No. of advertisement:  10
No. of wepbage:  9


In [ ]:
# Convert datetime features
ctr_df["datetime"] = pd.to_datetime(ctr_df["DateTime"])

# **Overview of the data**

In [ ]:
ad_df = ctr_df.groupby(["product", "campaign_id"]).count()["user_id"].reset_index()
ad_df["campaign_id"] = ad_df["campaign_id"].astype(str)
fig = px.bar(ad_df,
             x="product",
             y="user_id",
             color="campaign_id",
             text_auto=True)

fig.update_layout(template="plotly_white", title="Product in Campaign Ads")
fig.show()

- In this viz, we can deduce that `campaign id` is not directly translated to a campaign ad. We can say that a `campaign id` represents who created the campaign, and can be different videos or visuals.
- Among the products offered, product `C` has the highest number of campaigns shows. With over 160k advertisement. This is followed by product `H` and `I` with almost 120k and over 60k respectively.
- The largest number of `campaign_id` in Products `A`, `C`, `F`, `H` is campaign id `405490`.
- The product with the lowest number of campaigns is product `F`, with almost 8k advertisement.

In [ ]:
web_df = ctr_df.groupby(["product", "webpage_id"]).count()["user_id"].reset_index()
web_df["webpage_id"] = web_df["webpage_id"].astype(str)
fig = px.bar(web_df,
             x="product",
             y="user_id",
             color="webpage_id",
             text_auto=True)

fig.update_layout(template="plotly_white", title="Product shows in Webpages")
fig.show()

- Product `C` has the highest number of presence among websites. Followed by product `H` and `I`.
- `website_id` 13787 has the highest number of ads for Product `C`, `H`, `I`, `B`, and `D`.

### **Successes in advertisement and exposure**

In [ ]:
ctr_df.columns

Index(['session_id', 'DateTime', 'user_id', 'product', 'campaign_id',
       'webpage_id', 'product_category_1', 'product_category_2',
       'user_group_id', 'gender', 'age_level', 'user_depth',
       'city_development_index', 'var_1', 'is_click', 'datetime'],
      dtype='object')

In [ ]:
ctr_adv = ctr_df.groupby(["product","campaign_id", "is_click"]).count()["user_id"].reset_index()
ctr_adv["campaign_id"] = ctr_adv["campaign_id"].astype(str)
ctr_adv["is_click"] = ctr_adv["is_click"].astype(str)

In [ ]:
fig = px.bar(ctr_adv,
             x="product",
             y="user_id",
             color="campaign_id",
             text_auto=True,
             facet_row="is_click")

fig.update_layout(template="plotly_white", title="Successes from Campaign Ads", barmode="stack")
fig.show()

In [ ]:
ctr_web = ctr_df.groupby(["product","webpage_id", "is_click"]).count()["user_id"].reset_index()
ctr_web["webpage_id"] = ctr_web["webpage_id"].astype(str)
ctr_web["is_click"] = ctr_web["is_click"].astype(str)

In [ ]:
fig = px.bar(ctr_web,
             x="product",
             y="user_id",
             color="webpage_id",
             text_auto=True,
             facet_row="is_click")

fig.update_layout(template="plotly_white", title="Successes from Web Ads", barmode="stack")
fig.show()

In [ ]:
ctr_date = ctr_df.groupby(['datetime', 'is_click']).count()["user_id"].reset_index()
ctr_date["is_click"] = ctr_date["is_click"].astype(str)
fig = px.line(ctr_date,
              x='datetime',
              y='user_id',
              color="is_click")
fig.update_layout(template="plotly_white")
fig.show()

/usr/local/lib/python3.10/dist-packages/_plotly_utils/basevalidators.py:105: FutureWarning:

The behavior of DatetimeProperties.to_pydatetime is deprecated, in a future version this will return a Series containing python datetime objects instead of an ndarray. To retain the old behavior, call `np.array` on the result



### **Click thru rate (CTR)**

In [ ]:
rate_ads = ctr_df.groupby(["campaign_id", "is_click"]).count()["user_id"].reset_index()
rate_ads

campaign_id  is_click  user_id
0         82320         0    26073
1         82320         1     1776
2         98970         0    32371
3         98970         1     2694
4        105960         0    24019
5        105960         1     1762
6        118601         0    33635
7        118601         1     1896
8        359520         0   101815
9        359520         1     6340
10       360936         0    49542
11       360936         1     2346
12       396664         0    23100
13       396664         1     1809
14       404347         0    26591
15       404347         1     2235
16       405490         0    87210
17       405490         1     8763
18       414149         0    27604
19       414149         1     1710